content/en/blog/_posts/2024-04-22-userns-beta/index.md

```diff
@@ -0,0 +1,156 @@
+namespace](https://man7.org/linux/man-pages/man7/user_namespaces.7.html). This
+namespace allows us to isolate the user and group identifiers (UIDs and GIDs) we
+use inside the container from the ones on the host.
+
+This is a powerful abstraction that allows us to run containers as "root": we
+are root inside the container and we can do everything root can inside the pod,
+but our interactions with the host are limited to what a non-privileged user can
+do. This is great to limit the impact of a container breakout.
+
+A container breakout is when a process inside a container is able to break out
+onto the host using some unpatched vulnerability in the container runtime or the
+kernel and able to access / modify files on the host or other containers. If we
+run our pods with user namespaces, the privileges the container has over the
+rest of the host are reduced and the files outside the container it can access
+are limited too.
+
+In Kubernetes v1.25 we introduced support for user namespaces for only stateless
+pods. Kubernetes 1.28 lifted that restriction and now, with Kubernetes 1.30, we
+are moving to beta!
+
+## What is a user namespace?
+
+Note: Linux user namespaces are a different concept from [kubernetes
+namespaces](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/).
+The former is a Linux kernel feature, the latter is a kubernetes feature.
+
+User namespaces is a Linux feature that isolates the UIDs and GIDs of the
+containers from the ones on the host. The identifiers in the container can be
+mapped to identifiers on the host in a way where the host UID/GIDs used for
+different containers never overlap. Furthermore, the identifiers can be mapped
+to unprivileged non-overlapping UIDs and GIDs on the host. This brings two key
+benefits:
+
+ * _Prevention of lateral movement_: As the UIDs and GIDs for different
+containers are mapped to different UIDs and GIDs on the host, containers have a
+harder time attacking each other even if they escape the container boundaries.
+For example, if container A is running with different UIDs and GIDs on the host
+than container B, the operations it can do on container B's files and processes
+are limited: only read/write what a file allows to others, as it will never
+have permission owner or group permission (the UIDs/GIDs on the host are
+guaranteed to be different for different containers).
+
+ * _Increased host isolation_: As the UIDs and GIDs are mapped to unprivileged
+users on the host, if a container escapes the container boundaries, even if it
+is running as root inside the container, it has no privileges on the host. This
+greatly protects what host files it can read/write, which process it can send
+signals to, etc. Furthermore, capabilities granted are only valid inside the
+user namespace and not on the host, which also limits the impact a container
+escape can have.
+
+![UIDs/GIDs layout for userns](userns-ids.png)
```

Comment:
_Improvement:_ To display the image incorporate a '[figure](https://gohugo.io/content-management/shortcodes/#figure)' shortcode with a caption and maybe also include an alt attribute for better accessibility. Take a look at how the shortcode is utilized in a [previous blog post (here)](https://github.com/kubernetes/website/blob/main/content/en/blog/_posts/2021-09-29-data-duplication-in-data-heavy-k8s-env.md?plain=1#L109).

Reply:
Improved that, thanks!